src/SmartComponents/FormComponents/BulkSelectButton.js

```diff
@@ -0,0 +1,99 @@
+import React, { useEffect, useState } from 'react';
+import { Button } from '@patternfly/react-core';
+import PropTypes from 'prop-types';
+import useFormApi from '@data-driven-forms/react-form-renderer/use-form-api';
+import useFieldApi from '@data-driven-forms/react-form-renderer/use-field-api';
+import { SELECT_ALL } from '../../Utilities/constants';
+import './BulkSelectButton.scss';
+
+// eslint-disable-next-line no-unused-vars
+const BulkSelectButton = (props) => {
+  const {
+    group,
+    section,
+    label,
+    input: { onChange, checked, ...input },
+  } = useFieldApi({
+    ...props,
+    type: 'button',
+  });
+  const formOptions = useFormApi();
+  const [buttonName, setButtonName] = useState(label);
+  useEffect(() => {
```

Comment:
the useEffect should have dependencies to run only when needed, not with each render. You can maybe use `dirtyFields` from formApi for that check